tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1037,6 +1038,34 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  /** Resolves any [[RecordType]] in a [[PreTransform]]. */
+  private def resolveRecordType(
+      preTrans: PreTransform): Option[(RecordType, CancelFun)] = {
```

Comment:
I have added more comments in the Scaladoc of that method to avoid that confusion.